components/brave_wallet/browser/cardano/cardano_transaction.h

```diff
@@ -0,0 +1,179 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_TRANSACTION_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_TRANSACTION_H_
+
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "brave/components/brave_wallet/browser/cardano/cardano_rpc_schema.h"
+#include "brave/components/brave_wallet/common/cardano_address.h"
+
+namespace brave_wallet {
+
+inline constexpr uint32_t kCardanoTxHashSize = 32u;
+inline constexpr uint32_t kCardanoWitnessSize = 96u;
+
+class CardanoTransaction {
+ public:
+  // Cardano tx outpoint. Pair of transaction id and its output index.
+  struct Outpoint {
+    Outpoint();
+    ~Outpoint();
+    Outpoint(const Outpoint& other);
+    Outpoint& operator=(const Outpoint& other);
```

Comment:
Is this absolutely necessary that these types in this file have copy contructors? I would prefer if they were move only.

Reply:
Copying is necessary

Reply:
Would you elaborate please.

Reply:
Wha I mean is that I need an example of code where it is being used. Could you shown me where that's the case.

Reply:
As explained, I need to understand if these copy constructors are the best option. This is part of the code review process. Please reply to my question above.

Reply:
@supermassive I'm looking at that example and I find it very hard to follow. To me it seems like you're making  unnecessary copies where you could use references. I suspect it has something to do with the fact that you move here: `cur_transaction = std::move(next_transaction);`, but it's not clear to me why you're doing that either.